sinks/api/types.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"time"
+
+	cadvisor "github.com/google/cadvisor/info"
+)
+
+type MetricType int
+
+const (
+	// A cumulative metric.
+	MetricCumulative MetricType = iota
+
+	// An instantaneous value metric.
+	MetricGauge
+)
+
+func (self MetricType) String() string {
+	switch self {
+	case MetricCumulative:
+		return "cumulative"
+	case MetricGauge:
+		return "gauge"
+	}
+	return ""
+}
+
+type MetricValueType int
+
+const (
+	// An int64 value.
+	ValueInt64 MetricValueType = iota
+	// A boolean value
+	ValueBool
+	// A double-precision floating point number.
+	ValueDouble
+)
+
+func (self MetricValueType) String() string {
+	switch self {
+	case ValueInt64:
+		return "int64"
+	case ValueBool:
+		return "bool"
+	case ValueDouble:
+		return "double"
+	}
+	return ""
+}
+
+type MetricUnitsType int
+
+const (
+	// A metric with no units.
+	UnitsNone MetricUnitsType = iota
```

Comment:
Ah, I think you want a Count type as well. That takes care of the None usecase above.